README.md

```diff
@@ -1,11 +1,173 @@
+- [Creating a New Simple Node](#creating-a-new-simple-node)
+- [Launch Files](#launch-files)
+- [Testing](#testing)
+  - [GTest](#gtest)
+  - [Rostest](#rostest)
+- [Running Packages](#running-packages)
+  - [Running UWSim](#to-run-uwsim)
+  - [Running Phidget IMU](#running-phidget)
+  - [Running Camera Driver](#cammera-driver)
 
-![alt tag](https://travis-ci.org/ubc-subbots/Jormungandr.svg?branch=master)
 
+## Setup 
+
+### Rosinstall
+
+Rostinstall is addressed and run in `install_dependencies.sh`, should there be any need to run manual, rosinstall should be run after all steps from installation and setup has been completed. 
+
+## Conventions
+
+### Github Conventions
+- We follow the Forking Workflow: here is what it is [here](https://www.atlassian.com/git/tutorials/comparing-workflows#forking-workflow) and how to use it [here](https://gist.github.com/Chaser324/ce0505fbed06b947d962)
+- Only commit files that are essential for the system to run; do not put any photos or videos in here
+- All files **must** be formatted properly. Formatting will be enforced with the `clang-format` tool. 
+    - To check and fix formatting, from the `Snowflake` folder run `./clang_format/fix_formatting.sh BRANCH_NAME`, where `BRANCH_NAME` is the name of the branch you intend to merge your code into (ex. `iarrc` or `core`). This script will fix any improperly formatted code, but will refuse to change any files with uncommited changes (to prevent you losing work)
+- Once your pull request has been reviewed and revised until it looks good from both your and the reviewers' sides, go ahead and Squash and Merge it, which will squash all the commits on your pull request into one and merge it to the target branch.
+
+### Coding Conventions
+- Every **.cpp** and **.h** file should start with 
+```
+/*
+ * Created By: Someone
+ * Created On: December 1st, 2000
+ * Description: A quick description of what this file does/is for
+ */
+```
+
+- Functions should be commented a la JavaDoc
+- The Javadoc comment (below) should be directly above every function in the header file
+```
+/**
+ * One line description of the function
+ *
+ * A longer and more in depth description of the function
+ * if it is needed.
+ * 
+ * @param param_one the first parameter of the function
+ * @param param_two the second parameter of the function whose
+ *                  description goes longer than one line
+ * @return what the function returns if it returns anything
+ * 
+ */
+```
+
+- Classes are **CamelCase**
+- Variables are **non_camel_case**
+- Constants are **ALL_CAPS_WITH_UNDERSCORES**
+- Functions are **camelCase**
+- Indentations are 4 spaces
+
+- `CMakeLists.txt` files should be reduced to only contain the minimum amount of comments. The version in `sample_package` has all the comments left in (for the sake of verbosity), so for a more representative example of what yours should look like, see `sb_vision/CMakeLists.txt` (or really any package aside from `sample_package`)
```

Comment:
`sb_vision` doesn't exist in this repo, but we could replace it with a link to the `sb_vision/CMakeLists.txt` in the Snowflake repo

Reply:
Done, edited